beacon_node/lighthouse_network/src/peer_manager/mod.rs

```diff
@@ -74,6 +78,16 @@ pub struct PeerManager<TSpec: EthSpec> {
     target_peers: usize,
     /// Peers queued to be dialed.
     peers_to_dial: VecDeque<(PeerId, Option<Enr>)>,
+    /// The number of temporarily banned peers. This is used to prevent instantaneous
+    /// reconnection.
+    // NOTE: This just prevents re-connections. The state of the peer is otherwise unaffected. A
+    // peer can be in a disconnected state and new connections will be refused and logged as if the
+    // peer is banned without it being reflected in the peer's state.
+    // Also the banned state can out-last the peer's reference in the peer db. So peers that are
+    // unknown to us can still be temporarily banned. This is fundamentally a relationship with
+    // the swarm. Regardless of our knowledge of the peer in the db, it will be temporarily banned
+    // at the swarm layer.
+    temporary_banned_peers: LRUTimeCache<PeerId>,
```

Comment:
Yes! I originally had this as a hashset_delay, but I then got concerned that we have lots of disconnections happening all the time. 
I think I was polling it in the peer_manager poll and thought it might not be ideal waking these tasks all the time. 
I opted for this cache, so that we just store these there without the need for a specific timely poll. Instead we piggy-back off the heartbeat timer. This means the time-resolution of when we unban these peers can only be as small as the heartbeat timeout. I was hoping that would be more efficient and sufficient for this purpose, so long as the temp ban is > heartbeat timeout. 
If it becomes less, then we practically just have the heartbeat timeout as the ban timeout. 

If you think there is negligible difference between the two, a `hashset_delay` would be the better choice as it gives us the unbanning at the exact time we set.